lib/blockchain-data-helper.ts

```diff
@@ -16,4 +24,91 @@ const fetchListingsFromBlockchain = async (count: number) => {
   return results;
 };
 
-export { fetchListingsFromBlockchain };
+const fetchAssetDataFromContract = async (
+  address: string,
+  userAddress: string,
+) => {
+  const mint = prepareEvent({
+    signature: "event Mint(address,uint256)",
+  });
+  const events = await getContractEvents({
+    contract: monetPointsContractFactory(address),
+    fromBlock: "earliest",
+    toBlock: "latest",
+    events: [mint],
+  });
+  const decimals = await readContract({
+    contract: monetPointsContractFactory(address),
+    method: "decimals",
+  });
+  const pointSymbol = async () => {
+    const Symboldata = await readContract({
+      contract: monetPointsContractFactory(address),
+      method: "symbol",
+    });
+    return Symboldata;
+  };
+  const pointName = async () => {
+    const Namedata = await readContract({
+      contract: monetPointsContractFactory(address),
+      method: "name",
+    });
+    return Namedata;
+  };
+  const asset = async () => {
+    const Assetdata = await readContract({
+      contract: monetMarketplaceContract,
+      method: "getAsset",
+      params: [address],
+    });
+    return Assetdata;
+  };
+
+  const userOnChainPoints = async () => {
+    const points = await readContract({
+      contract: monetPointsContractFactory(address),
+      method: "balanceOf",
+      params: [userAddress],
+    });
+    return points;
+  };
+
+  const calculateMintedPoints = async () => {
+    let mintedPoints = 0;
+    const userEvents = events.filter((event) => event.args[0] === userAddress);
+    userEvents.forEach((event) => {
+      mintedPoints += Number(toTokens(BigInt(event.args[1]), decimals));
```

Comment:
We should avoid using Number when dealing with points and other things from smart contracts. Can we make it using BigInt only ?